controller/volume_controller.go

```diff
@@ -1054,18 +1054,25 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		aggregatedVerboseErrors = util.AppendVerboseErrors(aggregatedVerboseErrors, verboseErrors)
 
 		if scheduledReplica == nil {
+			verboseErrorMessage := replicaSchedulingFailureMessage(aggregatedVerboseErrors)
+			if verboseErrorMessage == "" {
+				verboseErrorMessage = longhorn.SchedulerErrorSchedulingFailed
+			}
```

Comment:
I would recommend aggregating and setting the scheduling failure message outside of the `for` loop.
The logic may be a little bit tricky. VolumeController will mark the volume scheduling condition as False **as long as** there is **one** replica failing at it. But the error message is the **summarization** of **all** scheduling failures and should be set after the iteration.

Reply:
Yeah, indeed this is tricky, but I feel the intermediate messages should be fine but leave it to @derekbit to decide.